realm/app-manager/src/key/hkdf.rs

```diff
@@ -0,0 +1,112 @@
+use std::collections::HashSet;
+
+use hkdf::Hkdf;
+use rust_rsi::RSI_SEALING_KEY_FLAGS_SVN;
+use sha2::digest::typenum::Unsigned;
+use sha2::{digest::OutputSizeUser, Sha256};
+use thiserror::Error;
+
+use super::Result;
+use crate::config::RsiSealingKeyFlags;
+use crate::{config::IkmSource, consts::APPLICATION_SLK_SALT};
+
+use super::{KeySealing, KeySealingFactory};
+
+#[derive(Debug, Error)]
+pub enum HkdfSealingError {
+    #[error("Failed to decode hex in stubbed ikm")]
+    StubbedIkmDecodingError(#[source] const_hex::FromHexError),
+
+    #[error("Failed to fetch source key material from RSI")]
+    IkmRsiReadingError(#[source] rust_rsi::NixError),
+}
+
+pub struct HkdfSealing {
+    ikm: Vec<u8>,
+    use_image_hash: bool,
+}
+
+impl KeySealing for HkdfSealing {
+    fn derive_key(&self, infos: &[&[u8]]) -> crate::error::Result<Vec<u8>> {
+        let hkdf = Hkdf::<Sha256>::new(None, &self.ikm);
+        let mut okm = [0u8; <Sha256 as OutputSizeUser>::OutputSize::USIZE];
+
+        hkdf.expand_multi_info(infos, &mut okm)
+            .expect("OKM size mismatch");
```

Comment:
I don't think this returning an error is required here as the used hash function is SHA256 and cannot change based on user input. Therefore, this will only fail when the HKDF library changes its behavior, which is very unlikely as it implements an NIST cryptographic standard.